### PR TITLE
[FEAT] Couchbase as remote connector

### PR DESCRIPTION
**Pull request by wizenheimer**
## Summary
- Add Couchbase adapter and connector implementation for storage backend
- Implement CouchbaseAdapter with get/put/contains operations 
- Add CouchbaseConnector with connection management and configuration
- Add comprehensive tests for both adapter and connector functionality
- Update connector __init__.py to expose new Couchbase classes

## Changes
- **New file**: `lmcache/v1/storage_backend/connector/couchbase_adapter.py` - Core Couchbase operations adapter
- **New file**: `lmcache/v1/storage_backend/connector/couchbase_connector.py` - Couchbase connection management
- **Modified**: `lmcache/v1/storage_backend/connector/__init__.py` - Export new Couchbase classes
- **Modified**: `tests/conftest.py` - Add Couchbase test fixtures and configuration
- **Modified**: `tests/v1/test_connector.py` - Add comprehensive Couchbase connector tests

## Test plan
- [x] Unit tests for CouchbaseAdapter operations (get, put, contains)
- [x] Unit tests for CouchbaseConnector connection management
- [x] Integration tests with mocked Couchbase cluster
- [x] Error handling and edge case testing
- [x] Configuration validation testing

---

<details>

<summary><b> PR Checklist (Click to Expand) </b></summary>

<p>Thank you for your contribution to LMCache! Before submitting the pull request, please ensure the PR meets the following criteria. This helps us maintain the code quality and improve the efficiency of the review process.</p>

<h3>PR Title and Classification</h3>
<p>Please try to classify PRs for easy understanding of the type of changes. The PR title is prefixed appropriately to indicate the type of change. Please use one of the following:</p>
<ul>
    <li><code>[Bugfix]</code> for bug fixes.</li>
    <li><code>[CI/Build]</code> for build or continuous integration improvements.</li>
    <li><code>[Doc]</code> for documentation fixes and improvements.</li>
    <li><code>[Model]</code> for adding a new model or improving an existing model. Model name should appear in the title.</li>
    <li><code>[Core]</code> for changes in the core LMCache logic (e.g., <code>LMCacheEngine</code>, <code>Backend</code> etc.)</li>
    <li><code>[Misc]</code> for PRs that do not fit the above categories. Please use this sparingly.</li>
</ul>
<p><strong>Note:</strong> If the PR spans more than one category, please include all relevant prefixes.</p>

<h3>Code Quality</h3>

<p>The PR need to meet the following code quality standards:</p>

<ul>
    <li>The code need to be well-documented to ensure future contributors can easily understand the code.</li>
    <li> Please include sufficient tests to ensure the change is stay correct and robust. This includes both unit tests and integration tests.</li>
</ul>

<h3>What to Expect for the Reviews</h3>

To create a new tag for lmcache (Note: `v` prefix is required):
`git tag vx.x.x`
`git push origin vx.x.x` (same version again)

For example:
`git tag v0.3.0`
`git push origin v0.3.0`

In case the workflow fails, delete the tag and try again:
`git tag -d vx.x.x`
`git push origin :refs/tags/vx.x.x`

For example:
`git tag -d v0.3.0`
`git push origin :refs/tags/v0.3.0`

To create a new release and publish `lmcache` Python package to PyPi:
`git remote add upstream git@github.com:LMCache/LMCache.git`
`gh release create vx.x.x --repo LMCache/LMCache --title "vx.x.x" --notes "<Add description>"`

For example:
`git remote add upstream git@github.com:LMCache/LMCache.git`
`gh release create v0.3.0 --repo LMCache/LMCache --title "v0.3.0" --notes "LMCache v0.3.0 is a feature release. Users are encouraged to upgrade for the best experience."`

> [!TIP]
> The creation of a release and subsequent tag generation can be done alternatively from the LMCache [releases](https://github.com/LMCache/LMCache/releases) page.

We aim to address all PRs in a timely manner. If no one reviews your PR within 5 days, please @-mention one of KuntaiDu, ApostaC or YaoJiayi.

### CODE REVIEW

File: lmcache/v1/storage_backend/connector/couchbase_adapter.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+class CouchbaseConnectorAdapter(ConnectorAdapter):
+    """Adapter for Couchbase connectors."""
+
+    def __init__(self) -> None:
+        super().__init__("couchbase://")
+
+    def can_parse(self, url: str) -> bool:
+        """Check if this adapter can parse the given URL."""
+        return url.startswith(self.schema) or url.startswith("couchbases://")
```
Comment (sammshen) [inline lmcache/v1/storage_backend/connector/couchbase_adapter.py:25]: you can pass in a tuple here: 
```python
url.startswith((self.schema, "couchbases://"))

File: lmcache/v1/storage_backend/connector/couchbase_connector.py [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+            options = ClusterOptions(auth)
+            self.cluster = Cluster(connection_string, options)
+        else:
+            self.cluster = Cluster(connection_string)
+
+        self.bucket = self.cluster.bucket(self.bucket_name)
+        self.scope = self.bucket.scope(self.scope_name)
+        self.collection = self.scope.collection(self.collection_name)
+
+    def _get_key_string(self, key: CacheEngineKey) -> str:
```
Comment (sammshen) [inline lmcache/v1/storage_backend/connector/couchbase_connector.py:100]: not sure if this wrapper is useful

**Comment by sammshen** [comment]
thanks for adding unit tests. 

1. you seemed to fail your own unit tests? 

```text

        # Test 4: Verify key exists after putting data
        future = asyncio.run_coroutine_threadsafe(connector.exists(random_key), async_loop)
        assert future.result()
>       assert memory_obj.get_ref_count() == 1
E       assert 0 == 1
E        +  where 0 = get_ref_count()
E        +    where get_ref_count = <lmcache.v1.memory_management.TensorMemoryObj object at 0x7fd6e01cff20>.get_ref_count
tests/v1/test_connector.py:121: AssertionError


```

2. can you run this E2E?

File: lmcache/v1/storage_backend/connector/couchbase_connector.py [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+
+    async def exists(self, key: CacheEngineKey) -> bool:
+        """Check if key exists in Couchbase."""
+        key_str = self._get_key_string(key)
+        metadata_key = key_str + METADATA_KEY_SUFFIX
+
+        try:
+            # Run in thread pool since Couchbase SDK is synchronous
+            result = await self.loop.run_in_executor(
+                None, lambda: self.collection.exists(metadata_key)
```
Comment (sammshen) [inline lmcache/v1/storage_backend/connector/couchbase_connector.py:112]: why not: 
```python
await self.loop.run_in_executor(None, self.collection.exists, metadata_key)
```

sorry for the nitpicky comments btw, just curious

File: lmcache/v1/storage_backend/connector/couchbase_connector.py [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+                # Store metadata
+                kv_shape = memory_obj.get_shape()
+                kv_dtype = memory_obj.get_dtype()
+                memory_format = memory_obj.get_memory_format()
+
+                metadata_bytes = RemoteMetadata(
+                    len(kv_bytes), kv_shape, kv_dtype, memory_format
+                ).serialize()
+
+                await self.loop.run_in_executor(
```
Comment (sammshen) [inline lmcache/v1/storage_backend/connector/couchbase_connector.py:225]: will there be consistency issues in separating the metadata and the KV storage?

File: lmcache/v1/storage_backend/connector/couchbase_connector.py (hunk 3 above)
Comment (sammshen) [reply lmcache/v1/storage_backend/connector/couchbase_connector.py:112]: same issue here, if exists oly check metadata key, you will not be alerted of eviction sof the real data. can CouchBase create some kind of PubSub for eviction events?
